### ISSUE TITLE
False positive on missing native type hint for inherited class properties

### ISSUE THREAD

**Issue by wazum** [squizlabs/PHP_CodeSniffer#3591]
**Describe the bug**

Code Sniffer returns an error 
```
Property \Vendor\Something\Something::$someProperty does not have native type hint for its value but it should be possible to add it based on @var annotation "array<string>".
```
for a property of an inherited class without a native type hint.

**Code sample**
```php
// Can't change that
class ExternalVendorPackageClass {
    protected $someProperty = [];
}
class MyLocalPackageClass extends ExternalVendorPackageClass {
    // leads to Code Sniffer error
    /** @var array<string> */
    protected $someProperty = [
        'customA',
        'customB'
    ];
}
```

**Expected behavior**
No error

**Versions**
 - PHP: 8.1
 - PHPCS: 3.6.2 (stable)

**Comment by jrfnl** [comment]
That is not a message coming from one of the PHPCS native sniffs. I suspect it is coming from the [SlevomatCodingStandard](https://github.com/slevomat/coding-standard/), but can't be sure as the issue template was not filled out correctly/completely.

I suggest you run PHPCS with the `-s` option to see the error code which should confirm the standard this error is coming from and that you open an issue there.

Suggest: close

**Comment by wazum** [comment]
Right, thanks
```
SlevomatCodingStandard.TypeHints.PropertyTypeHint.MissingNativeTypeHint
```
I'll look there for further suggestions …
